internal/handler/msg_updateuser.go

```diff
@@ -96,11 +96,51 @@ func (h *Handler) MsgUpdateUser(ctx context.Context, msg *wire.OpMsg) (*wire.OpM
 		return nil, lazyerrors.Error(err)
 	}
 
+	if mechanisms == nil {
+		return nil, handlererrors.NewCommandErrorMsg(
+			handlererrors.ErrBadValue,
+			"mechanisms field must not be empty",
+		)
+	}
+
+	if mechanisms.Len() == 0 {
+		return nil, handlererrors.NewCommandErrorMsg(
+			handlererrors.ErrBadValue,
+			"mechanisms field must not be empty",
+		)
```

Comment:
This condition is new and not tested by integration test. Shouldn't we have an integration test if we change logic?

Reply:
I removed this as well as it's unnecessary, it will default to `defMechanisms` when missing.